### ISSUE TITLE
RxJavaAssemblyTracking impacts execution and produce different results

### ISSUE THREAD

**Issue by freszu** [akarnokd/RxJava2Extensions#30]
Recently I came across weird bug which occurred only in debug version of the app that has assembly tracking enabled. 
It occurs when combination of ```replay(1).refcount()``` is used, resulting in a weird bug where first subscription after dispose is not emitting anything while second works as expected.

I was able to isolate the problem and narrow it down to simple example in Java:
```Java
public class Showcase {
    public static void main(String[] args) throws InterruptedException {
        RxJavaAssemblyTracking.enable();

        System.out.println("start");

        PublishSubject<String> stringsEmitter = PublishSubject.create();

        Observable<String> combineSource = stringsEmitter
                .replay(1)
                .refCount();

        CompositeDisposable c = new CompositeDisposable();

        c.add(
                combineSource
                        .subscribeOn(Schedulers.io())
                        .subscribe((string) -> System.out.println("A1:" + string))
        );
        c.add(
                combineSource
                        .subscribeOn(Schedulers.io())
                        .subscribe((string) -> System.out.println("B1:" + string))
        );

        stringsEmitter.onNext("s1");

        Thread.sleep(100);
        c.clear();

        combineSource.subscribeOn(Schedulers.io())
                .subscribe((string) -> System.out.println("A2:" + string));

        combineSource.subscribeOn(Schedulers.io())
                .subscribe((string) -> System.out.println("B2:" + string));

        Thread.sleep(100);

        stringsEmitter.onNext("s2");

        Thread.sleep(1000);

        stringsEmitter.onNext("s3");
        stringsEmitter.onNext("s4");
    }
}
```
Output i get when assembly tracker is disabled - as expected:
```
start
A2:s2
B2:s2
A2:s3
B2:s3
A2:s4
B2:s4
```
Output when assembly tracker is enabled:
```
start
B2:s2
B2:s3
B2:s4
```

RxJava version 2.2.7
RxJava2Extensions 0.20.6

**Comment by akarnokd** [comment]
Hi and thanks for reporting. It really does seem quite odd and you don't need scheduling; A2 will not receive s2, s3 and s4 for some reason. I'll investigate this further soon.

**Comment by akarnokd** [comment]
This was indeed a bug in the tracking wrapper not forwarding the reset option for the `replay` operator. Fix in [4b929e21](https://github.com/akarnokd/RxJava2Extensions/commit/4b929e21a4efaa1aaf20dbc57fb48ade1910dd5f). I'll make a release once the changes clear.
